src/common/closh/zero/builtin.cljc

```diff
@@ -45,8 +46,15 @@
   "Changes current working directory to a path of a first given argument."
   [& args]
   ;; flatten is used because we can get arguments from expand which are collections
-  (let [dir (or (first args)
-                (getenv "HOME"))]
+  (let [arg      (first args)
+        go-back? (= "-" arg)
+        dir      (or (if go-back?
+                       (getenv "OLDPWD")
+                       arg)
+                     (getenv "HOME"))]
+    (setenv "OLDPWD" (process/cwd))
     (process/chdir (str dir)) ; Extra (str ..) to handle case when directory name is a number
     (setenv "PWD" (process/cwd))
+    (when go-back?
+      (core/shx "pwd" [] {:redir [[:set 0 :stdin] [:set 2 :stderr] [:set 1 :stdout]]}))
```

Comment:
Do we need this or is it left over from debugging?

Reply:
We do need this.
According to the docs `cd -` should behave as `cd $OLDPWD && pwd` (besides settings env) - thus we need to print pwd (maybe in a different way tho).
(https://pubs.opengroup.org/onlinepubs/9699919799/utilities/cd.html
ref on this found herehttps://unix.stackexchange.com/questions/330876/difference-between-cd-and-cd)

Also, it seems a bit tricky how it can be tested (should I mock stdout for each env?)

Please advice how you think it can be tested.

Thanks.

Reply:
I see, that's a good point. Traditional shells print to stdout, but I think we could try to just return the new `PWD` value instead of `env/success`. It will still get printed out when used on the command line. And it should be easier to test.

Reply:
hey @dundalek ,

returning `PWD` indeed prints it, but it is printed via `prn` thus string is quoted, like this: `"/Users/kirill.salykin/projects/closh"`. (which is not the way how other shells do this)

This is seems done by rebel.

What do you think? 

Thanks.

Reply:
It may be fixed with something like this:

```
(defn repl-print
  [& args]
  (if (string? (first args))
    (println (first args))
    (when-not (or (nil? (first args))
                  (identical? (first args) env/success)
                  (process? (first args)))
      (apply syntax-highlight-prn args))))
```
Shall I go for this?

Reply:
I think I kinda like it that it is printed as a string, feels more structured. I think it is good for now, if it causes any trouble in the future we can change it to println.